firmware/controllers/sensors/impl/software_knock.cpp

```diff
@@ -55,38 +55,46 @@ static const uint32_t smpr2 =
 	ADC_SMPR2_SMP_AN8(KNOCK_SAMPLE_TIME) |
 	ADC_SMPR2_SMP_AN9(KNOCK_SAMPLE_TIME);
 
-static const ADCConversionGroup adcConvGroupCh1 = { FALSE, 1, &completionCallback, &errorCallback,
-	0,
-	ADC_CR2_SWSTART,
+static const ADCConversionGroup adcConvGroupCh1 = {
+	.circular = FALSE,
+	.num_channels = 1,
+	.end_cb = &completionCallback,
+	.error_cb = &errorCallback,
+	.cr1 = 0,
+	.cr2 = ADC_CR2_SWSTART,
 	// sample times for channels 10...18
-	smpr1,
+	.smpr1 = smpr1,
 	// sample times for channels 0...9
-	smpr2,
+	.smpr2 = smpr2,
 
-	0,	// htr
-	0,	// ltr
+	.htr = 0,
+	.ltr = 0,
 
-	0,	// sqr1
-	0,	// sqr2
-	ADC_SQR3_SQ1_N(KNOCK_ADC_CH1)
+	.sqr1 = 0,
+	.sqr2 = 0,
+	.sqr3 = ADC_SQR3_SQ1_N(KNOCK_ADC_CH1)
 };
 
 // Not all boards have a second channel - configure it if it exists
 #if KNOCK_HAS_CH2
-static const ADCConversionGroup adcConvGroupCh2 = { FALSE, 1, &completionCallback, &errorCallback,
-	0,
-	ADC_CR2_SWSTART,
-	// sample times for channels 10...18
-	smpr1,
-	// sample times for channels 0...9
-	smpr2,
-
-	0,	// htr
-	0,	// ltr
-
-	0,	// sqr1
-	0,	// sqr2
-	ADC_SQR3_SQ1_N(KNOCK_ADC_CH2)
+static const ADCConversionGroup adcConvGroupCh2 = {
+    .circular = FALSE,
+   	.num_channels = 1,
```

Comment:
strange indent?

Reply:
@mck1117 should be better now

Reply:
ok, indents were _not_ better, something is very wrong with your editor

![image](https://github.com/FOME-Tech/fome-fw/assets/568254/c9e677bb-53a5-47f4-bbbb-b5e9b11c7cf1)

Reply:
> ok, indents were _not_ better, something is very wrong with your editor

my editor might be missing an .editorconfig in this repo?

Reply:
> my editor might be missing an .editorconfig in this repo?

There are lots of funny spots in rusEFI too.

![image](https://github.com/FOME-Tech/fome-fw/assets/568254/5f065cc7-dc35-4d70-9835-ed87e450df94)